src/ansible/roles/keycloak/tasks/main.yml

```diff
@@ -0,0 +1,110 @@
+- name: Create Keycloak user/group
+  user:
+    name: keycloak
+    home: /opt/keycloak
+    system: yes
+    create_home: yes
+
+- name: Print keycloak_url
+  debug:
+    msg: "keycloak_url={{ keycloak_url }}"
+
+- name: Download software zipfile
+  get_url:
+    url: "{{ keycloak_url }}"
+    dest: /tmp
+
+- name: Unzip software
+  unarchive:
+    remote_src: yes
+    src: "/tmp/{{ zip_file }}"
+    dest: /opt/keycloak
+    owner: keycloak
+    group: keycloak
+    extra_opts:
+      - --strip-components=1
+
+- name: Change ownership of files in /data/certs
+  file:
+    path: "/data/certs/{{ item }}"
+    owner: keycloak
+    group: keycloak
+  loop:
+    - master.keycloak.test.crt
+    - master.keycloak.test.key
```

Comment:
crt is already readable by everyone, I think we can make key readable as well in common.

Reply:
that would render this step unnecessary, unless you need the ownership.

Reply:
I'll double check the ownership and permissions requirement.

Reply:
@pbrezina 

I tried changing all key file permissions to 644 but, that breaks the ipa trust-add to the samba domain.   It makes more sense to me to adjust the ownership or permissions for the file in the keycloak role since that's the only one that requires a different setting.  

Do you still prefer I change it in common or can it stay in the keycloak role?   Or do you have another suggestion?

Reply:
Ok, if it breaks, then keep it.